user: Set CONSTRUCT flag for properties where appropriate

melix99 opened the pull request:
This way we can reliably do unwrap on properties that we don't want to be nullable.

This should probably be done in other places too, but I will do that in another PR.

melix99 committed:
user: Set CONSTRUCT flag for properties where appropriate

This way we can reliably do unwrap on properties that we don't want to be nullable.

marhkb commented on src/session/user.rs:
```diff
@@ -22,10 +22,10 @@ mod imp {
+        pub first_name: RefCell<Option<String>>,
+        pub last_name: RefCell<Option<String>>,
+        pub username: RefCell<Option<String>>,
+        pub phone_number: RefCell<Option<String>>,
```
What's the advantage of wrapping Strings in Options?

melix99 replied:
Let's say that this makes more sense since we don't need having a default String value because it will be replaced anyway when constructing the GObject.

melix99 replied:
I know that in the end doing String::new() is still not doing allocations, but using Options is more consistent with the other objects and we can also differenciate between unset property and empty string property.

marhkb replied:
How are we going to handle other types? For example, if we want to be consistent, we are also forced to wrap primitive types like integers.

melix99 replied:
For primitive types it makes even less sense, so I don't plan on doing that.

melix99 replied:
So basically I'd like to follow the rule of:
- Primitive types => `Cell<>`
- Objects => `RefCell<Option<>>`

marhkb replied:
> ... we don't need having a default String value ...

I was referring to that statement. If we don't need a default string value of "" then we probably also don't need a default integer value of `0`. :-)

melix99 replied:
Well, yeah, that point actually makes less sense since glib now should (hopefully) cover us about setting that value in the constructor before using it around. So, the remaining point is the consistency :p

marhkb replied:
And I am also not sure what's your definition of an Object. 
Are that heap allocated types? Or is it the size of a type?
For example, this works fine in a `Cell`
```rust
#[derive(Copy, Clone)]
struct Large([u8; 10000000]);

fn main() {
    let t = std::cell::Cell::new(Large([1; 10000000]));
}
```

melix99 replied:
> And I am also not sure what's your definition of an Object.

I meant a more complex type than the primitive ones.

melix99 committed:
user: Correctly handle string properties

Strings in GObjects are nullable, so we have to expect Option<String> instead of just String. Also, update the default values to be Some("") instead of None.
